src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -208,11 +215,21 @@ private static void RunServer(string serverPort, string semaphoreName, int clien
                             ServiceName,
                             WellKnownObjectMode.Singleton);
 
-                        var uiThread = new Thread(UIThread);
-                        uiThread.SetApartmentState(ApartmentState.STA);
-                        uiThread.IsBackground = true;
-                        uiThread.Start();
-                        s_uiReady.Wait();
+                        using (var resetEvent = new ManualResetEventSlim(false))
+                        {
+                            var uiThread = new Thread(() =>
+                            {
+                                var c = new Control();
+                                c.CreateControl();
+                                s_UIThreadScheduler = TaskScheduler.FromCurrentSynchronizationContext();
```

Comment:
I'm not sure I understand the interaction between the static and instance parts of this class.  Presumably, there's reason to believe that s_UIThreadScheduler will be null before this is called (so as not to clobber the previous value) and won't be used during that time?